tests/modelwrapper_test.py

```diff
@@ -99,6 +100,29 @@ def test_predict_on_batch(self):
         pred = self.wrapper.predict_on_batch(input, 10, False)
         assert pred[0].size() == (2, 1, 10)
 
+        # iteration == 1
+        self.wrapper = ModelWrapper(self.model, self.criterion, replicate_in_memory=False)
+        pred = self.wrapper.predict_on_batch(input, 1, False)
+        assert pred[0].size() == (2, 1, 1)
+
+        # iterations > 1
+        pred = self.wrapper.predict_on_batch(input, 10, False)
+        assert pred[0].size() == (2, 1, 10)
+
+    def test_out_of_mem_raises_error(self):
```

Comment:
related to the above - we should maybe add a test here that checks we raise a RuntimeError as you have & that when a "different" error (maybe build a model that always raises a ValueError) occurs, we raise that error and not the RuntimeError that we've got up there

Reply:
Sounds good. Raising a TypeError would be pretty easy, that should cover for that second case I guess?

Reply:
perfect!